MetabolicCompass/View Controller/AuthorizationViewControllers/RegisterViewController.swift

```diff
@@ -104,29 +104,30 @@ private let inputFontSize = ScreenManager.sharedInstance.profileInputFontSize()
     func doConsent() {
         stashedUserId = UserManager.sharedManager.getUserId()
         UserManager.sharedManager.resetFull()
-        ConsentManager.sharedManager.checkConsentWithBaseViewController(self.navigationController!) { [weak self] consentAndNames -> Void in
-            guard consentAndNames.0 else {
+        ConsentManager.sharedManager.checkConsentWithBaseViewController(self.navigationController!) { (consent, firstName, lastName) in
+            guard consent else {
                 UserManager.sharedManager.resetFull()
-                if let user = self!.stashedUserId {
+                if let user = self.stashedUserId {
                     UserManager.sharedManager.setUserId(userId: user)
                 }
-                self!.navigationController?.popViewController(animated: true)
-                return
+                self.navigationController?.popViewController(animated: true)
+                return ()
             }
 
             // Note: add 1 to index, due to photo field.
-            if let s = self {
-                let updatedData = consentAndNames.1 != nil || consentAndNames.2 != nil
-                if consentAndNames.1 != nil { s.dataSource.model.setAtItem(itemIndex: s.dataSource.model.firstNameIndex+1, newValue: consentAndNames.1! as AnyObject?) }
-                if consentAndNames.2 != nil { s.dataSource.model.setAtItem(itemIndex: s.dataSource.model.lastNameIndex+1, newValue: consentAndNames.2! as AnyObject?) }
-                if updatedData { s.collectionView.reloadData() }
+            if self != nil {
```

Comment:
`self` never will be `nil`, since this block retains it
until we add `[weak self]` in capture list